src/camera.rs

```diff
@@ -86,18 +90,28 @@ impl Camera {
     ) -> Option<(Color, f32)> {
         let dist = center - self.transform.position;
         let a = ray.sq_mag();
+        // SAFETY: `a` will always be positive; we let LLVM know so this can be optimized.
+        unsafe {
+            std::intrinsics::assume(a >= 0.0);
+        }
+
         let b = ray.dot(dist);
         let c = dist.sq_mag() - r.powi(2);
 
-        let sqrt_term = (b.powi(2) - a * c).sqrt();
-        if !sqrt_term.is_finite() {
+        let sqrt_term_inner = b.powi(2) - a * c;
+        if 0.0 > sqrt_term_inner || sqrt_term_inner.is_subnormal() {
```

Comment:
The mathematical name for the value inside the square root term is "discriminant," so let's use that.
```suggestion
        let discriminant = b.powi(2) - a * c;
        if discriminant.is_sign_negative() || discriminant.is_subnormal() {
```